src/allmydata/test/test_storage_http.py

```diff
@@ -694,7 +722,6 @@ def test_mismatching_upload_fails(self):
                     b"0123456789",
                 )
             )
-            self.assertEqual(e.exception.code, http.NOT_FOUND)
```

Comment:
This assertion was never reached (because the previous function raised an exception) and moreover was wrong.